botocore/handlers.py

```diff
@@ -441,8 +447,33 @@ def switch_host_with_param(request, param_name):
      base64_encode_user_data),
     ('before-parameter-build.route53', fix_route53_ids),
     ('before-parameter-build.glacier', inject_account_id),
+
+    # Glacier documentation customizations
     ('docs.*.glacier.*.complete-section',
-     AutoPopulatedParam('accountId').document_auto_populated_param),
+     AutoPopulatedParam('accountId', 'Note: this parameter is set to "-" by \
+                         default if no value is not specified.')
+     .document_auto_populated_param),
     ('docs.*.glacier.*.complete-section',
-     AutoPopulatedParam('checksum').document_auto_populated_param)
+     AutoPopulatedParam('checksum').document_auto_populated_param),
+    # UserData base64 encoding documentation customizations
+    ('docs.*.ec2.RunInstances.complete-section', document_base64_encoding()),
+    ('docs.*.autoscaling.CreateLaunchConfiguration.complete-section',
+     document_base64_encoding()),
+    # EC2 CopySnapshot documentation customizations
+    ('docs.*.ec2.CopySnapshot.complete-section',
+     AutoPopulatedParam('PresignedUrl').document_auto_populated_param),
+    ('docs.*.ec2.CopySnapshot.complete-section',
+     AutoPopulatedParam('DestinationRegion').document_auto_populated_param),
+    # S3 SSE documentation modifications
+    ('docs.*.s3.*.complete-section',
+     AutoPopulatedParam('SSECustomerKeyMD5').document_auto_populated_param),
+    # The following S3 operations cannot actually accept a ContentMD5
+    ('docs.*.s3.*.complete-section',
+     HideParamFromOperations(
+         's3', 'ContentMD5',
+         ['DeleteObjects', 'PutBucketAcl', 'PutBucketCors',
```

Comment:
The following operations do not have a documented Content-MD5 header:

- PutBucketAcl: http://docs.aws.amazon.com/AmazonS3/latest/API/RESTBucketPUTacl.html
- PutBucketLogging: http://docs.aws.amazon.com/AmazonS3/latest/API/RESTBucketPUTlogging.html
- PutBucketNotification: http://docs.aws.amazon.com/AmazonS3/latest/API/RESTBucketGETnotification.html
- PutBucketPolicy: http://docs.aws.amazon.com/AmazonS3/latest/API/RESTBucketPUTpolicy.html
- PutBucketRequestPayment: http://docs.aws.amazon.com/AmazonS3/latest/API/RESTrequestPaymentPUT.html
- PutBucketVersioning: http://docs.aws.amazon.com/AmazonS3/latest/API/RESTBucketPUTVersioningStatus.html
- PutBucketWebsite: http://docs.aws.amazon.com/AmazonS3/latest/API/RESTBucketPUTwebsite.html
- PutObjectAcl: http://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectPUTacl.html

These operations can, however, take a Content-MD5 header as described in S3's common request headers documentation: http://docs.aws.amazon.com/AmazonS3/latest/API/RESTCommonRequestHeaders.html.

Thing is, you don't need to add a Content-MD5 check for these operations when using SignatureV4 because the message integrity check is built-in to the sigv4 signature. I think this means that we should either not do anything for these operations (other than hide them in docs) or add a new handler that only adds a Content-MD5 header if the signature version is not sigv4.

Adding a new handler would be a new handler and a separate PR I think.

Thoughts?